lib/optimizely.rb

```diff
@@ -263,23 +263,26 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
       end
 
       decision = @decision_service.get_variation_for_feature(feature_flag, user_id, attributes)
-      unless decision.nil?
-        variation = decision['variation']
-        # Send event if Decision came from an experiment.
-        if decision.source == Optimizely::DecisionService::DECISION_SOURCE_EXPERIMENT
-          send_impression(decision.experiment, variation['key'], user_id, attributes)
-        else
-          @logger.log(Logger::DEBUG,
-                      "The user '#{user_id}' is not being experimented on in feature '#{feature_flag_key}'.")
+      if decision.nil?
+        @logger.log(Logger::INFO,
+                    "Feature '#{feature_flag_key}' is not enabled for user '#{user_id}'.")
+        return false
+      end
+      variation = decision['variation']
+      if decision.source == Optimizely::DecisionService::DECISION_SOURCE_EXPERIMENT
+        unless variation['featureEnabled']
```

Comment:
This check shouldn't be here. We also use `featureEnabled` for rollout variations.